### ISSUE TITLE
Support more components

### ISSUE THREAD

**Issue by dogukanteber** [cortexproject/auth-gateway#19]
In the config file, we currently support the distributor and the query-frontend components. As @songjiayang suggested in [this issue](https://github.com/cortexproject/auth-gateway/issues/10#issuecomment-1519621015), we should support alertmanager and ruler components as well. After #6 is merged, the necessary modifications need to be made in the config file.

**Comment by songjiayang** [comment]
Another question，how can we use it to proxy other services, like a cortex extend services,  for example cortex-slo( a slos genertor serivce).

**Comment by friedrichg** [comment]
@songjiayang That is a great question. I agree it should be possible to use auth-gateway with other services. Maybe not configured by default in any way, but possible.

**Comment by songjiayang** [comment]
There is an [example](( https://github.com/telemetrytower/cortex-gateway/blob/main/cortex-gateway.yaml)) about our cortex-gateway config file.

**Comment by dogukanteber** [comment]
This issue is closed but can be reopened or another issue can be created if necessary.

Is your request satisfied with the latest changes @songjiayang? You can create a configuration by [this example configuration](https://github.com/cortexproject/auth-gateway/blob/main/gateway/testdata/configuration.yaml). Note that the given configuration does not contain ruler and alertmanager but the current implementation supports those components. You can configure these components as you desire.

The configuration you provided can be rewritten like below to work with `auth-gateway`:

```yaml

server:
  address: localhost
  port: 8080
  http_server_read_timeout: 20s
  http_server_write_timeout: 15s
  http_server_idle_timeout: 90s
admin:
  address: localhost
  port: 8081
  http_server_read_timeout: 20s
  http_server_write_timeout: 15s
tenants:
  - authetication: basic
  - username: <type your username>
  - password: <type your password>
  - id: <org id>
distributor:
   url: http://127.0.0.1:8004
   paths:
   -  /api/v1/push
frontend:
   url: http://127.0.0.1:8004
   paths:
   - /prometheus
ruler:
  url: http://127.0.0.1:8004
  paths:
   - /api/v1/alerts
   - /api/v1/rules
alertmanager:
   url: http://127.0.0.1:8004
   paths:
   - /alertmanager
```

I did not put `secret` key here because I do not know its purpose. You can define custom timeouts for each component and can change the server configuration as you please.

I would like to answer your questions if you have any.

**Comment by songjiayang** [comment]
@dogukanteber thanks  for your patiently explain. 

`secret` in our gateway is used for Bearer authorization.

**Comment by dogukanteber** [comment]
My pleasure @songjiayang.

Currently, `auth-gateway` only supports simple authentication. If you would like to use `auth-gateway` with Bearer authorization, feel free to open an issue. That way we can see if other people would like to see Bearer authorization implemented too.
